elements/pfe-navigation/src/pfe-navigation.js

```diff
@@ -0,0 +1,335 @@
+
+if (!Array.prototype.filter){
+  Array.prototype.filter = function(func, thisArg) {
+    'use strict';
+    if ( ! ((typeof func === 'Function' || typeof func === 'function') && this) )
+        throw new TypeError();
+   
+    var len = this.length >>> 0,
+        res = new Array(len), // preallocate array
+        t = this, c = 0, i = -1;
+
+    var kValue;
+    if (thisArg === undefined){
+      while (++i !== len){
+        // checks to see if the key was set
+        if (i in this){
+          kValue = t[i]; // in case t is changed in callback
+          if (func(t[i], i, t)){
+            res[c++] = kValue;
+          }
+        }
+      }
+    }
+    else{
+      while (++i !== len){
+        // checks to see if the key was set
+        if (i in this){
+          kValue = t[i];
+          if (func.call(thisArg, t[i], i, t)){
+            res[c++] = kValue;
+          }
+        }
+      }
+    }
+   
+    res.length = c; // shrink down array to proper size
+    return res;
+  };
+}
+
+class PfeNavigation extends PFElement {
+  static get tag() {
+    return "pfe-navigation";
+  }
+
+  get templateUrl() {
+    return "pfe-navigation.html";
+  }
+
+  get styleUrl() {
+    return "pfe-navigation.scss";
+  }
+
+  get schemaUrl() {
+    return "pfe-navigation.json";
+  }
+
+  get overlay() {
+    return !this._overlay.hasAttribute("hidden");
+  }
+
+  set overlay(state) {
+    if (state) {
+      // Add the overlay to the page
+      this._overlay.removeAttribute("hidden");
+      // This prevents background scroll while nav is open
+      document.body.style.overflow = "hidden";
+    } else {
+      // Remove the overlay from the page
+      this._overlay.setAttribute("hidden", "");
+      // Allow background to scroll again
+      document.body.style.overflow = "auto";
+
+    }
+  }
+
+  constructor() {
+    super(PfeNavigation);
+
+    // Attach functions for use below
+    this._init = this._init.bind(this);
+    this._setVisibility = this._setVisibility.bind(this);
+
+    // -- handlers
+    this._keydownHandler = this._keydownHandler.bind(this);
+    this._resizeHandler = this._resizeHandler.bind(this);
+    this._stickyHandler = this._stickyHandler.bind(this);
+    this._outsideListener = this._outsideListener.bind(this);
+    this._observer = new MutationObserver(this._init);
+
+    // Capture shadow elements
+    this._overlay = this.shadowRoot.querySelector(".pfe-navigation__overlay");
+    this._menuItem = this.shadowRoot.querySelector(".pfe-navigation__main > pfe-navigation-item");
+
+    this._slots = {
+      language: this.shadowRoot.querySelector(".pfe-navigation__main--mobile > pfe-navigation-item:first-of-type"),
+      login: this.shadowRoot.querySelector(".pfe-navigation__main--mobile > pfe-navigation-item:last-of-type")
+    };
+
+    // Initialize active navigation item to empty array
+    this._activeNavigationItems = [];
+    this.overlay = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
```

Comment:
We should probably make sure the other components, `pfe-navigation-item` and `pfe-navigation-main`, have been created before we get into the `this.children.length` check. 

Follow `pfe-tabs` as an example.
https://github.com/patternfly/patternfly-elements/blob/master/elements/pfe-tabs/src/pfe-tabs.js#L157

Reply:
:+1: Added this.